Documentation/Relationship/Relationship.md

```diff
@@ -0,0 +1,334 @@
+
+## 2. Manually defining "structural" relationship information
+
+This chapter imagines ways how a user could manually override or provide additional relationship information.
+
+### 2.1. Overriding the relationship name
+
+The above described inferring approach will use the string description of the `PathComponent` as the relationship name.
+This may not fit everybody's needs.
+
+Similar to how you can manually define a `@Parameter` name, you can also override a relationship name.
+This proposal suggest a sort of `PathComponentModifier` (similar to `ComponentModifiers`).
+
+Such a definition could then look like the following:
+```swift
+var content: some Component {
+  Group("test".relationshipName("new-name")) {
+    Handler()
+  }
+}
+```
+
+### 2.2. Adding new relationships
+
+In order to support rearranging components, a user can also define its own relationships.
+We introduce a `.relationship` modifier for `PathComponent`s, to which one would pass
+a relationship name and `PathComponent`s describing the destination.  
+The destination can of course also include `@PathParameters`.
+
+Any manual definition of a relationship MUST always override an inferred relationship if the name collides.
+
+Looking at the example below, a REST exporter would add the link `greeter` pointing to `/greeting/{userid}` on a 
+response returned from `/user/:userId`. As the value of the `userId` `PathParameter` is known, it is replaced with
+the actual value. `@PathParameter`s for which no predefined value can be inferred should be formatted according to
+[RFC 6570](https://tools.ietf.org/html/rfc6570) (as well as potential query parameters of the destination). 
+
+```swift
+var content: some Component {
+  Group("user", $userId.relationship("greeter", to: "greeting", $userId)) {
+    Handler()
+  }
+  Group("greeting", $userId) {
+    Greeter()
+  }
+}
+```
+
+### 2.3. Excluding relationships
+
+It may also be a sensible decision for a user to hide a inferred relationship. To support that
+a modifier can be introduce.
+
+Even if a user supplies a `.relationshipName` modifier, the relationship will still be hidden when `.hideRelationship` is used.
+
+Considering the example below, a request to `/` will not include a relationship pointing to `/test`.
+
+```swift
+var content: some Component {
+  Group("test".hideRelationship()) {
+    Handler()
+  }
+}
+```
+
+## 3. Using type information
+
+Checking the structure of the Webservice is already pretty powerful, but doesn't allow for relationships between handlers
+which are on the same level or on different paths of the path component tree.  
+This chapter highlights how we can nonetheless retrieve relationship information by looking at type information.
+
+### 3.1 Looking at the return type of `handle()`
+
+For this chapter we will look at the following example web service:
+
+```swift
+struct User: Identifiable {
```

Comment:
This should also be `Codable`? right?

Reply:
Out of the scope of this proposal. But yeah, whatever makes it able to be encoded and decoded.